crates/router/src/services/api.rs

```diff
@@ -606,7 +610,8 @@ where
         .authenticate_and_fetch(request.headers(), state)
         .await
         .switch()?;
-    let metric_merchant_id = auth_out.get_merchant_id().unwrap_or("").to_string();
+    let merchant_id = auth_out.get_merchant_id().unwrap_or("").to_string();
```

Comment:
Are you suggesting that we use some value which `merchant_id` cannot take? We currently have no validation for `merchant_id`, so it can take any value. We should add validations to merchant id ( do not contain spaces, no special characters, have atleast 5 or more characters ), In that case having the `merchant_id` as empty in logs makes sense as `merchant_id` cannot take the value of `""` after validations.